planex/cmd/fetch.py

```diff
@@ -121,11 +122,10 @@ def fetch_http(url, filename, retries):
         url_string, headers=headers, timeout=30, stream=True)
     req.raise_for_status()
 
-    tmp_filename = filename + "~"
-    with open(tmp_filename, 'wb') as out:
+    with tempfile.NamedTemporaryFile(delete=False) as out:
         shutil.copyfileobj(req.raw, out)
-    best_effort_file_verify(tmp_filename)
-    shutil.move(tmp_filename, filename)
+    best_effort_file_verify(out.name)
+    shutil.move(out.name, filename)
```

Comment:
I think you can just get rid of the temporary file altogether and write to the final destination.   If the verification fails, exit with a non-zero exit code and `make` will delete it.   If you run `planex-fetch` by hand you could end up with a 'bad' file but that could actually be an advantage because you can then see what you actually downloaded (often a redirect or error page from GitHub) without the need for another `--keeptmp` flag.

I think we added this tempfile initially because the makefile was trying to download the same file several times in parallel.  We should treat that as a bug in the makefile.

Reply:
Good, let's get rid of that then